openvdb_houdini/houdini/SOP_OpenVDB_Sample_Points.cc

```diff
@@ -346,20 +402,86 @@ VDB_NODE_OR_CACHE(VDB_COMPILABLE_SOP, SOP_OpenVDB_Sample_Points)::cookVDBSop(OP_
         const GU_Detail* bGdp = inputGeo(1, context); // where the grids live
 
         // extract UI data
-        const bool verbose = (evalInt("verbose", 0, time) != 0);
+        const bool verbose = evalInt("verbose", 0, time) != 0;
         const bool threaded = true; /*evalInt("threaded", 0, time);*/
+
+        // total number of points in vdb grids - this is used when verbose execution is
+        // requested, but will otherwise remain 0
+        size_t nVDBPoints = 0;
+        std::vector<std::string> includeGroups;
+        std::vector<std::string> excludeGroups;
+        UT_String vdbPointsGroups;
+
+        // obtain names of vdb point groups to include / exclude
+
+        evalString(vdbPointsGroups, "vdbpointsgroups", 0, time);
+
+        cvdb::points::AttributeSet::Descriptor::parseNames(includeGroups, excludeGroups,
+            vdbPointsGroups.toStdString());
+
+        // extract VDB points grids
+
+        std::vector<cvdb::points::PointDataGrid::Ptr> pointGrids;
+
+        for (openvdb_houdini::VdbPrimIterator it(gdp); it; ++it) {
+            GU_PrimVDB* vdb = *it;
+
+            if (!vdb) continue;
+            if (!vdb->getConstGridPtr()->isType<cvdb::points::PointDataGrid>()) continue;
+
+            vdb->makeGridUnique();
+
+            cvdb::GridBase::Ptr grid = vdb->getGridPtr();
+            cvdb::points::PointDataGrid::Ptr pointDataGrid =
+                cvdb::gridPtrCast<cvdb::points::PointDataGrid>(grid);
+
+            if (verbose) {
+
+                auto leaf = pointDataGrid->tree().cbeginLeaf();
+
+                if (leaf) {
+                    cvdb::points::MultiGroupFilter filter(includeGroups, excludeGroups,
+                        leaf->attributeSet());
+
+                    nVDBPoints +=
+                        cvdb::points::filterPointCount<cvdb::points::PointDataTree,
+                            cvdb::points::MultiGroupFilter>(pointDataGrid->tree(), filter);
+                }
+            }
+
+            pointGrids.push_back(pointDataGrid);
+        }
```

Comment:
For the sake of simplifying the code, it might be cleaner to just change the return value of points::sampleGrid() to return the number of points sampled so that this logic could be removed entirely. That could have the added benefit of extracting the point count changes into a new PR to be done independently.